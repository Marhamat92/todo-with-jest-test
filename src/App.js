import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { SafeAreaView, StyleSheet, Text, TextInput, View } from 'react-native';
import Button from './components/Button/Button';
import tw from 'twrnc';  
import { FlatList } from 'react-native';


export default function App() {

  const [tasks, setTasks] = React.useState([]);
  const [task, setTask] = React.useState('');

const renderElements = ({item}) => <Text style={tw`text-black `}>{item}</Text>
 

const addTask = () => {
 if(task.length > 0){
  setTasks([...tasks,task]);}
  setTask('');
}


  return (
    <SafeAreaView>
    <View style={tw`mt-20 flex `}>
      <Text>Testing</Text>
      <FlatList
      keyExtractor={(item, index) => index.toString()}
      testID='flatlist'
      data={tasks}
      renderItem={renderElements}
      />
      <TextInput onChangeText={setTask}  testID='input-area' placeholder='add your task here..' />
      <Button onClick={addTask} title="Add Task"/>
       
    </View>
    </SafeAreaView>
  );
}


