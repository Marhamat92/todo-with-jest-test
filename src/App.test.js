import React from 'react'
import {render,fireEvent} from '@testing-library/react-native'
import App from './App'

test('should match with snapshot',()=>{
    const comp = render(<App/>)
    expect(comp).toMatchSnapshot()
})


test('should add a task',()=>{
    const comp = render(<App/>)
    const input = comp.getByTestId('input-area')
    fireEvent.changeText(input,'test element')


    const submitButton = comp.getByTestId('button-touchable')
    fireEvent.press(submitButton)

    const flatlist = comp.getByTestId('flatlist')
    expect(flatlist.children).toHaveLength(1)
})

test('should add tasks for multiple times',()=>{
    const comp = render(<App/>)
    const input = comp.getByTestId('input-area')
    fireEvent.changeText(input,'test element')

    const submitButton = comp.getByTestId('button-touchable')
    fireEvent.press(submitButton)
    fireEvent.press(submitButton)
    fireEvent.press(submitButton)
    fireEvent.press(submitButton)

    const flatlist = comp.getByTestId('flatlist')
    expect(flatlist.data.length).toBe(4)

})

test('should not add task if input area is empty',()=>{
    const comp = render(<App/>)
    const input = comp.getByTestId('input-area')
    fireEvent.changeText(input,'')

    const submitButton = comp.getByTestId('button-touchable')
    fireEvent.press(submitButton)

    const flatlist = comp.getByTestId('flatlist')
    expect(flatlist.data.length).toBe(0)
})

//npm run test
//npm test -- -u
//npm test -- -u --watch
//npm test -- -u --watchAll
//npm test -- -u --watchAll --verbose
//npm test -- -u --watchAll --verbose --color
//npm test -- -u --watchAll --verbose --color --verbose
//npm test --coverage