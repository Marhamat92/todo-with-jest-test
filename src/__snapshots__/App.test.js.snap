// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "display": "flex",
        "marginTop": 80,
      }
    }
  >
    <Text>
      Testing
    </Text>
    <RCTScrollView
      data={Array []}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      testID="flatlist"
      viewabilityConfigCallbackPairs={Array []}
    >
      <View />
    </RCTScrollView>
    <TextInput
      onChangeText={[Function]}
      placeholder="add your task here.."
      testID="input-area"
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#22c55e",
          "borderRadius": 6,
          "marginTop": 8,
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 32,
          "paddingRight": 32,
          "paddingTop": 8,
        }
      }
      testID="button-touchable"
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 18,
            "fontWeight": "bold",
            "lineHeight": 28,
          }
        }
        testID="button-title"
      >
        Add Task
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
